src/interceptors/ClientRequest/ClientRequestOverride.ts

```diff
@@ -249,25 +249,18 @@ export function createClientRequestOverrideClass(
       let req: ClientRequest
       debug('using', performOriginalRequest)
 
-      const { ClientRequest } = http
-
       // Decide whether to use HTTPS based on the URL protocol.
       // XHR can trigger http.request for HTTPS URL.
       if (url.protocol === 'https:') {
         debug('reverting patches...')
+        const { ClientRequest } = http
 
         http.ClientRequest = originalClientRequest
 
-        // Override the global pointer to the original client request.
-        // This way whenever a bypass call bubbles to `handleRequest`
-        // it always performs respecting this `ClientRequest` restoration.
-        originalClientRequest = null as any
-
         req = performOriginalRequest(options)
 
         debug('re-applying patches...')
         http.ClientRequest = ClientRequest
```

Comment:
Should we restore this reference to the patched `ClientRequest` class when the `req` is actually finished (either successfully or unsuccessfully)?

I'm thinking that `performOriginalRequest` creates a request, but may provision references to `ClientRequest` between the moment the request instance is created and the moment the request is actually finished. On this line we restore the `http.ClientRequest` to the patched version, so if an underlying logic attempts to reference `ClientRequest` during the _original_ request, it would still point to the patched instance. 

That being said, how likely it is to reference `ClientRequest`? I don't think that's a pattern to create a request instance within a request instance. Especially, since that is how NodeJS operates, so you don't normally have access to this. My concern is valid only in corner cases like:

```js
// usage.js
// This request will be constructed using the original unpatched ClientRequest class.
const req = http.get('https://api.com/user', (res) => {
  // This request, however, will be constructed using the patched class, as we restored the ref.
  const anotherReq = http.get('https://another.domain')
})
```

In a way this is actually intentional, as that second request in a callback function should also be intercepted and cam potentially have a mocked response. Thank you for bearing with me through this comment 🦆

Reply:
http.get and http.request will create an instance of the ClientRequest when one of them is called so after performOriginalRequest we can restore the ClientRequest